# Copyright (C) 2010 Yegor Yelizarov <yegor.yelizarov@gmail.com>
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.


diff -ruN a/dsplink_linux_1_65_00_02/packages/dsplink/gpp/src/api/Linux/_notify.c b/dsplink_linux_1_65_00_02/packages/dsplink/gpp/src/api/Linux/_notify.c
--- a/dsplink_linux_1_65_00_02/packages/dsplink/gpp/src/api/Linux/_notify.c	2010-05-28 18:33:47.000000000 +0400
+++ b/dsplink_linux_1_65_00_02/packages/dsplink/gpp/src/api/Linux/_notify.c	2010-07-06 18:59:20.538862542 +0400
@@ -320,10 +320,12 @@
         pthread_exit (NULL) ;
     }
 
+#ifndef _ANDROID_
     if (pthread_sigmask (SIG_BLOCK, &blockSet, NULL) != 0) {
         perror ("Event worker thread error in setting sigmask") ;
         pthread_exit (NULL) ;
     }
+#endif
 
     drvHandle = open (LINK_DRIVER, O_RDWR | O_SYNC) ;
     if (drvHandle < 0) {
diff -ruN a/dsplink_linux_1_65_00_02/packages/dsplink/gpp/src/Makefile_kbuild b/dsplink_linux_1_65_00_02/packages/dsplink/gpp/src/Makefile_kbuild
--- a/dsplink_linux_1_65_00_02/packages/dsplink/gpp/src/Makefile_kbuild	2010-05-28 18:33:48.000000000 +0400
+++ b/dsplink_linux_1_65_00_02/packages/dsplink/gpp/src/Makefile_kbuild	2010-07-06 18:59:20.538862542 +0400
@@ -119,6 +119,7 @@
 	echo Kernel side build complete
 	echo Copying dsplinkk.ko to $(TI_DSPLINK_GPPROOT)$(DIRSEP)export$(DIRSEP)BIN$(DIRSEP)$(TI_DSPLINK_GPPOSPREFIX)Linux$(DIRSEP)$(TI_DSPLINK_PLATFORM)$(DIRSEP)DEBUG
 	mv dsplinkk.ko Module.symvers PMGR_includes.txt PMGR_defines.txt  $(TI_DSPLINK_GPPROOT)$(DIRSEP)export$(DIRSEP)BIN$(DIRSEP)$(TI_DSPLINK_GPPOSPREFIX)Linux$(DIRSEP)$(TI_DSPLINK_PLATFORM)$(DIRSEP)DEBUG
+	$(MAKE) PROFILE=DEBUG -C api
 
 release:
 	$(shell echo $(addprefix -I, $(TI_DSPLINK_GPP_INC_PATH)) > PMGR_includes.txt)
@@ -128,6 +129,7 @@
 	echo Kernel side build complete
 	echo Copying dsplinkk.ko to $(TI_DSPLINK_GPPROOT)$(DIRSEP)export$(DIRSEP)BIN$(DIRSEP)$(TI_DSPLINK_GPPOSPREFIX)Linux$(DIRSEP)$(TI_DSPLINK_PLATFORM)$(DIRSEP)RELEASE
 	mv dsplinkk.ko Module.symvers PMGR_includes.txt PMGR_defines.txt  $(TI_DSPLINK_GPPROOT)$(DIRSEP)export$(DIRSEP)BIN$(DIRSEP)$(TI_DSPLINK_GPPOSPREFIX)Linux$(DIRSEP)$(TI_DSPLINK_PLATFORM)$(DIRSEP)RELEASE
+	$(MAKE) PROFILE=RELEASE -C api
 
 clean:
 	find $(DSPLINK) -name "*.o" -exec rm {} \;
